stdlib/ballerina-grpc/src/main/java/org/ballerinalang/net/grpc/GrpcServicesBuilder.java

```diff
@@ -0,0 +1,277 @@
+import org.ballerinalang.net.grpc.listener.UnaryMethodListener;
+import org.ballerinalang.net.grpc.proto.ServiceProtoConstants;
+import org.ballerinalang.net.grpc.proto.ServiceProtoUtils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.ballerinalang.net.grpc.builder.BalGenConstants.FILE_SEPARATOR;
+
+/**
+ * This is the gRPC implementation for the {@code BallerinaServerConnector} API.
+ *
+ * @since 0.96.1
+ */
+public class GrpcServicesBuilder {
+    private io.grpc.ServerBuilder serverBuilder = null;
+    
+    void registerService(Service service) throws GrpcServerException {
+        Annotation serviceAnnotation = MessageUtils.getServiceConfigAnnotation(service, MessageConstants
+                .PROTOCOL_PACKAGE_GRPC);
+        if (serviceAnnotation != null && serviceAnnotation.getAnnAttrValue("port") != null) {
+            serverBuilder = NettyServerBuilder.forPort((int)
+                    serviceAnnotation.getAnnAttrValue("port").getIntValue())
+                    .bossEventLoopGroup(new NioEventLoopGroup(Runtime.getRuntime()
+                            .availableProcessors()))
+                    .workerEventLoopGroup(new NioEventLoopGroup(Runtime.getRuntime()
+                            .availableProcessors() * 2));
+        } else {
+            serverBuilder = NettyServerBuilder.forPort(9090)
+                    .bossEventLoopGroup(new NioEventLoopGroup(Runtime.getRuntime()
+                            .availableProcessors()))
+                    .workerEventLoopGroup(new NioEventLoopGroup(Runtime.getRuntime()
+                            .availableProcessors() * 2));
+        }
+        try {
+            serverBuilder.addService(ServerInterceptors.intercept(getServiceDefinition(service), new
+                    ServerHeaderInterceptor()));
+        } catch (Descriptors.DescriptorValidationException | GrpcServerException e) {
+            throw new GrpcServerException("Error while registering the service : " + service.getName(), e);
+        }
+    }
+    
+    private ServerServiceDefinition getServiceDefinition(Service service) throws Descriptors
+            .DescriptorValidationException, GrpcServerException {
+        Descriptors.FileDescriptor fileDescriptor = ServiceProtoUtils.getDescriptor(service);
+        Descriptors.ServiceDescriptor serviceDescriptor = fileDescriptor.findServiceByName(service.getName());
+        
+        Annotation serviceAnnotation = MessageUtils.getServiceConfigAnnotation(service, MessageConstants
+                .PROTOCOL_PACKAGE_GRPC);
+        if (isStreamService(serviceAnnotation)) {
+            return getStreamingServiceDefinition(service, serviceDescriptor);
+        } else {
+            return getUnaryServiceDefinition(service, serviceDescriptor);
+        }
+    }
+    
+    private boolean isStreamService(Annotation serviceAnnotation) {
+        if (serviceAnnotation == null) {
```

Comment:
Isn't it more readable to move this null check to higher layer rather handling it here? Similarly in other cases.